llama_index/indices/multi_modal/base_multi_modal_retriever.py

```diff
@@ -0,0 +1,95 @@
+import asyncio
+from typing import List
+
+from llama_index.indices.base_retriever import BaseRetriever
+from llama_index.indices.query.schema import QueryBundle, QueryType
+
+# from llama_index.constants import DEFAULT_SIMILARITY_TOP_K, DEFAULT_IMAGE_SIMILARITY_TOP_K
+from llama_index.retrievers.image_retriever import BaseImageRetriever
+from llama_index.schema import NodeWithScore
+
+
+class MultiModalRetriever(BaseRetriever, BaseImageRetriever):
+    """Multi Modal base retriever."""
+
+    def retrieve(self, str_or_query_bundle: QueryType) -> List[NodeWithScore]:
+        if isinstance(str_or_query_bundle, str):
+            str_or_query_bundle = QueryBundle(str_or_query_bundle)
+        return self._retrieve(str_or_query_bundle)
+
+    # Retrieve text and image nodes and mixer them
+    def _retrieve(
```

Comment:
Should we even define `_retrieve()` and `_aretrieve()` here? It's a little opinionated to assume someone would want to concat both text and image retrieval